src/config.py

```diff
@@ -0,0 +1,217 @@
+    protocol_fee_config: ProtocolFeeConfig
+    buffer_accounting_config: BufferAccountingConfig
+    io_config: IOConfig
+
+
+def get_accounting_config(network: str) -> AccountingConfig:
+    """Get config for specified network."""
+
+    # pylint: disable=too-many-locals
+
+    project_root_dir = Path(__file__).parent.parent
+    file_out_dir = project_root_dir / Path("out")
+    log_config_file = project_root_dir / Path("logging.conf")
+    query_dir = project_root_dir / Path("queries")
+    dashboard_dir = project_root_dir / Path("dashboards/solver-rewards-accounting")
+
+    docs_url = "https://www.notion.so/cownation/Solver-Payouts-3dfee64eb3d449ed8157a652cc817a8c"
+
+    # Secrets
+
+    # Found this exposed infura key on https://rpc.info/
+    infura_key = os.environ.get("INFURA_KEY", "9aa3d95b3bc440fa88ea12eaa4456161")
+
+    dune_api_key = os.environ.get("DUNE_API_KEY", "")
+
+    prod_db_url = os.environ.get("PROD_DB_URL", "")
+    barn_db_url = os.environ.get("BARN_DB_URL", "")
+
+    signing_key = os.getenv("PROPOSER_PK")
+    if signing_key == "":
+        signing_key = None
+
+    slack_channel = os.getenv("SLACK_CHANNEL", None)
+    slack_token = os.getenv("SLACK_TOKEN", None)
+
+    node_url = f"https://{network}.infura.io/v3/{infura_key}"
+
+    #
+    payment_safe_address = Web3.to_checksum_address(
+        os.environ.get("SAFE_ADDRESS", "0xA03be496e67Ec29bC62F01a428683D7F9c204930")
+    )
+    short_name = {
+        "mainnet": "eth",
+        "rinkeby": "rin",
+        "gnosis": "gno",
+        "goerli": "gor",
+    }[network]
+    csv_app_hash = "Qme49gESuwpSvwANmEqo34yfCkzyQehooJ5yL7aHmKJnpZ"
+    safe_url = f"https://app.safe.global/{short_name}:{payment_safe_address}"
+    airdrop_url = (
+        f"{safe_url}/apps?appUrl=https://cloudflare-ipfs.com/ipfs/{csv_app_hash}/"
+    )
+    safe_queue_url = f"{safe_url}/transactions/queue"
+
+    result = AccountingConfig(
+        payment_config=PaymentConfig(
+            network=EthereumNetwork.MAINNET,
+            cow_token_address=Address("0xDEf1CA1fb7FBcDC777520aa7f396b4E015F497aB"),
+            payment_safe_address=Web3.to_checksum_address(
+                "0xA03be496e67Ec29bC62F01a428683D7F9c204930"
+            ),
+            signing_key=signing_key,
+            safe_queue_url=safe_queue_url,
+            csv_airdrop_url=airdrop_url,
+            verification_docs_url=docs_url,
+            weth_address=Web3.to_checksum_address(
+                "0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2"
+            ),
+        ),
+        orderbook_config=OrderbookConfig(
+            prod_db_url=prod_db_url,
+            barn_db_url=barn_db_url,
+        ),
+        dune_config=DuneConfig(dune_api_key=dune_api_key, dune_blockchain="ethereum"),
+        node_config=NodeConfig(node_url=node_url),
+        reward_config=RewardConfig(
+            reward_token_address=Address("0xDEf1CA1fb7FBcDC777520aa7f396b4E015F497aB"),
+            batch_reward_cap_upper=12 * 10**15,
```

Comment:
I think we can make this easier to maintain as well by moving these configs into the RewardConfig class itself and passing the network onto the RewardConfig and setting these properties there based on the value for network.

> reward_config=RewardConfig("mainnet")

Reply:
I moved all the initializations into their respective classes. Most are `from_network` static method, some are `from_env` static methods.